crates/sui-graphql-rpc/src/types/execution_result.rs

```diff
@@ -1,14 +1,28 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use super::event::Event;
+use super::transaction_block_effects::TransactionBlockEffects;
 use async_graphql::*;
+use sui_json_rpc_types::BalanceChange as NativeBalanceChange;
+use sui_types::{
+    effects::TransactionEffects as NativeTransactionEffects,
+    transaction::SenderSignedData as NativeSenderSignedData,
+};
 
 #[derive(SimpleObject, Clone)]
 pub(crate) struct ExecutionResult {
     /// The errors field captures any errors that occurred during execution
     pub errors: Option<Vec<String>>,
 
-    /// The digest field captures the digest of the transaction block
-    pub digest: String,
-    // TODO: add support for `TransactionBlockEffects` field
+    /// The effects of the executed transaction.
+    pub effects: TransactionBlockEffects,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct ExecutedTransaction {
+    pub sender_signed_data: NativeSenderSignedData,
+    pub raw_effects: NativeTransactionEffects,
+    pub balance_changes: Vec<NativeBalanceChange>,
+    pub events: Vec<Event>,
```

Comment:
Realistically though this is a Vec of Native Event right, the stored field would be empty. If so I'd rather have this field be Vec<NativeEvent> as well

Reply:
Good point. Having a vec of native events here would be more consistent.